src/cdk/v2/destinations/intercom/utils.js

```diff
@@ -27,8 +31,30 @@ const {
   SEARCH_CONTACT_ENDPOINT,
   ReservedCompanyProperties,
   CREATE_OR_UPDATE_COMPANY_ENDPOINT,
+  TAGS_ENDPOINT,
 } = require('./config');
 
+const intercomErrorHandler = (message, processedResponse) => {
+  const errorMessages = JSON.stringify(processedResponse?.response?.errors);
+  if (processedResponse?.status === 400) {
+    throw new InstrumentationError(`${message} : ${errorMessages}`);
+  }
+  if (processedResponse?.status === 401) {
+    throw new ConfigurationError(`${message} : ${errorMessages}`);
+  }
+  if (processedResponse?.status === 404) {
+    throw new InstrumentationError(`${message} : ${errorMessages}`);
+  }
+  throw new NetworkError(
+    `${message} : ${errorMessages}`,
+    processedResponse?.status,
```

Comment:
ProcessedResponse is assumed to be present hence no need of optional chaining syntax like `processedResponse?.`